### PR TITLE
[PVR] Completely reworked playing flags. Moved almost all logic to PVR Manager

### PR DESCRIPTION
**Pull request by ksooo**
Another larger round of cleanup and simplification (which also should fix the bug recently reported by @xhaggi).

@FernetMenta what do you think about this?

I runtime-tested the changes on macOS, latest Kodi master.

### CODE REVIEW

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/addons/PVRClient.cpp [hunk 1]
```diff
@@ -1377,39 +1269,36 @@ PVR_ERROR CPVRClient::OpenStream(const CPVRChannelPtr &channel)
...
-PVR_ERROR CPVRClient::OpenStream(const CPVRRecordingPtr &recording)
+PVR_ERROR CPVRClient::OpenRecordedStream(const CPVRRecordingPtr &recording)
 {
   return DoAddonCall(__FUNCTION__, [this, recording](const AddonInstance* addon) {
-    CloseStream();
+    CloseRecordedStream();
 
     PVR_RECORDING tag;
     WriteClientRecordingInfo(*recording, tag);
+    CLog::Log(LOGDEBUG, "opening recorded '%s'", recording->m_strTitle.c_str());
```
Comment (xhaggi) [inline xbmc/addons/PVRClient.cpp]: `open recording` ?

**Comment by xhaggi** [comment]
Looks good .. thx @ksooo

**Comment by ksooo** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
I am not sure if the method of synchronizing states will work reliable. Why don't you use CApplication::m_currentItem as indicator? Keep in mind that we are going to split off the streaming method from PVR api. That means PVRManager should not depend on client's streaing methods and states.

**Comment by ksooo** [comment]
Well, this is actually what this PR is doing. PVR Manager no longer knows about playing states of the clients and feeds is own playing status information only from the `OnPlaying*` calls application does on PVR Manager when playing item changes.

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**